src/utils/helpers.js

```diff
@@ -0,0 +1,14 @@
+export function setButtonText(
+  btn,
+  isLoading,
+  loadingText = "Saving...",
+  defaultText = "Save"
+) {
+  if (isLoading) {
+    btn.textContent = loadingText;
+    btn.disabled = true;
+  } else {
+    btn.textContent = defaultText;
+    btn.disabled = false;
```

Comment:
`    btn.disabled = false;
`

Please, delete it from `setButtonText` because you **activate** the save button in `finally`, so it's impossible to keep the button disabled if you disable it in `then`